core/blockchain.go

```diff
@@ -301,7 +299,10 @@ func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, chainConfig *par
 	// The first thing the node will do is reconstruct the verification data for
 	// the head block (ethash cache or clique voting snapshot). Might as well do
 	// it in advance.
-	bc.engine.VerifyHeader(bc, bc.CurrentHeader(), true)
+	currentHeader := bc.CurrentHeader()
+	if err := bc.engine.VerifyHeader(bc, currentHeader, true); currentHeader.Number.Cmp(big.NewInt(0)) > 0 && err != nil {
```

Comment:
Rather than modifying the production code so that tests pass, I think it would be preferable to update the tests so that they run with valid data. Is that something we could do?

It seems that this code as is would allow for a blockchain instance to be constructed with an invalid genesis block.